### PR TITLE
added analytical-platform-image-build-template repo

### PR DESCRIPTION
**Pull request by Emterry**
- added terraform code
- create analytical-platform-image-build-template repo
- as per issue: #2502

### CODE REVIEW

File: terraform/github/analytical-platform-repositories.tf [hunk 1]
```diff
@@ -452,6 +452,20 @@ locals {
         admins  = [module.data_platform_teams["data-platform-apps-and-tools"].id]
         pushers = [module.data_platform_team.id]
       }
+    },
+    "analytical-platform-image-build-template" = {
+      name                                   = "analytical-platform-image-build-template"
+      description                            = "Analytical Platform Image Build Template"
+      is_template                            = true
+      use_template                           = false
```
Comment (julialawrence) [inline terraform/github/analytical-platform-repositories.tf]: We should use the default MOJ repo template for this.

File: terraform/github/analytical-platform-repositories.tf [hunk 2]
```diff
@@ -452,6 +452,20 @@ locals {
...
+      visibility                             = "public"
+      advanced_security_status               = "disabled"
```
Comment (julialawrence) [inline terraform/github/analytical-platform-repositories.tf]: Should we be turning this on or not needed for a template?

**Review by julialawrence — CHANGES_REQUESTED** [review]
Suggest using the moj repo template as a basis.

File: terraform/github/analytical-platform-repositories.tf (hunk 2 above)
Comment (Emterry) [reply terraform/github/analytical-platform-repositories.tf]: ive removed it as the docs says If a repository's visibility is public, advanced security is always enabled and cannot be changed

**Review by julialawrence — APPROVED** [review]
Good start. :)

**Pull request merged**